hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/helpers/OMAuditLogger.java

```diff
@@ -0,0 +1,188 @@
+    CMD_AUDIT_ACTION_MAP.put(Type.CancelDelegationToken, OMAction.CANCEL_DELEGATION_TOKEN);
+    CMD_AUDIT_ACTION_MAP.put(Type.RenewDelegationToken, OMAction.RENEW_DELEGATION_TOKEN);
+    CMD_AUDIT_ACTION_MAP.put(Type.GetS3Secret, OMAction.GET_S3_SECRET);
+    CMD_AUDIT_ACTION_MAP.put(Type.SetS3Secret, OMAction.SET_S3_SECRET);
+    CMD_AUDIT_ACTION_MAP.put(Type.RevokeS3Secret, OMAction.REVOKE_S3_SECRET);
+    CMD_AUDIT_ACTION_MAP.put(Type.CreateTenant, OMAction.CREATE_TENANT);
+    CMD_AUDIT_ACTION_MAP.put(Type.DeleteTenant, OMAction.DELETE_TENANT);
+    CMD_AUDIT_ACTION_MAP.put(Type.TenantAssignUserAccessId, OMAction.TENANT_ASSIGN_USER_ACCESSID);
+    CMD_AUDIT_ACTION_MAP.put(Type.TenantRevokeUserAccessId, OMAction.TENANT_REVOKE_USER_ACCESSID);
+    CMD_AUDIT_ACTION_MAP.put(Type.TenantRevokeAdmin, OMAction.TENANT_REVOKE_ADMIN);
+    CMD_AUDIT_ACTION_MAP.put(Type.CreateSnapshot, OMAction.CREATE_SNAPSHOT);
+    CMD_AUDIT_ACTION_MAP.put(Type.DeleteSnapshot, OMAction.DELETE_SNAPSHOT);
+    CMD_AUDIT_ACTION_MAP.put(Type.RenameSnapshot, OMAction.RENAME_SNAPSHOT);
+    CMD_AUDIT_ACTION_MAP.put(Type.RecoverLease, OMAction.RECOVER_LEASE);
+    CMD_AUDIT_ACTION_MAP.put(Type.CreateDirectory, OMAction.CREATE_DIRECTORY);
+    CMD_AUDIT_ACTION_MAP.put(Type.CreateFile, OMAction.CREATE_FILE);
+    CMD_AUDIT_ACTION_MAP.put(Type.CreateKey, OMAction.ALLOCATE_KEY);
+    CMD_AUDIT_ACTION_MAP.put(Type.AllocateBlock, OMAction.ALLOCATE_BLOCK);
+    CMD_AUDIT_ACTION_MAP.put(Type.CommitKey, OMAction.COMMIT_KEY);
+    CMD_AUDIT_ACTION_MAP.put(Type.DeleteKey, OMAction.DELETE_KEY);
+    CMD_AUDIT_ACTION_MAP.put(Type.DeleteKeys, OMAction.DELETE_KEYS);
+    CMD_AUDIT_ACTION_MAP.put(Type.RenameKey, OMAction.RENAME_KEY);
+    CMD_AUDIT_ACTION_MAP.put(Type.RenameKeys, OMAction.RENAME_KEYS);
+    CMD_AUDIT_ACTION_MAP.put(Type.InitiateMultiPartUpload, OMAction.INITIATE_MULTIPART_UPLOAD);
+    CMD_AUDIT_ACTION_MAP.put(Type.CommitMultiPartUpload, OMAction.COMMIT_MULTIPART_UPLOAD_PARTKEY);
+    CMD_AUDIT_ACTION_MAP.put(Type.AbortMultiPartUpload, OMAction.ABORT_MULTIPART_UPLOAD);
+    CMD_AUDIT_ACTION_MAP.put(Type.CompleteMultiPartUpload, OMAction.COMPLETE_MULTIPART_UPLOAD);
+    CMD_AUDIT_ACTION_MAP.put(Type.SetTimes, OMAction.SET_TIMES);
+    CMD_AUDIT_ACTION_MAP.put(Type.AbortExpiredMultiPartUploads, OMAction.ABORT_EXPIRED_MULTIPART_UPLOAD);
+    CMD_AUDIT_ACTION_MAP.put(Type.SetVolumeProperty, OMAction.SET_OWNER);
+    CMD_AUDIT_ACTION_MAP.put(Type.SetBucketProperty, OMAction.UPDATE_BUCKET);
+  }
+
+  private static OMAction getAction(OzoneManagerProtocolProtos.OMRequest request) {
+    Type cmdType = request.getCmdType();
+    OMAction omAction = CMD_AUDIT_ACTION_MAP.get(cmdType);
+    if (null != omAction) {
+      if (cmdType.equals(Type.SetVolumeProperty)) {
+        boolean hasQuota = request.getSetVolumePropertyRequest().hasQuotaInBytes();
+        if (hasQuota) {
+          return OMAction.SET_QUOTA;
+        }
+      }
+      if (cmdType.equals(Type.SetVolumeProperty)) {
```

Comment:
```suggestion
      if (cmdType.equals(Type.SetBucketProperty)) {
```
We can add another condition for set volume owner too

Reply:
updated